gnd/src/test/java/com/google/android/gnd/persistence/local/LocalDataStoreTest.java

```diff
@@ -0,0 +1,565 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.persistence.local;
+
+import static org.hamcrest.Matchers.samePropertyValuesAs;
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule;
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.LatLngBounds;
+import com.google.android.gnd.TestApplication;
+import com.google.android.gnd.inject.DaggerTestComponent;
+import com.google.android.gnd.model.Mutation;
+import com.google.android.gnd.model.Project;
+import com.google.android.gnd.model.User;
+import com.google.android.gnd.model.basemap.OfflineArea;
+import com.google.android.gnd.model.basemap.tile.Tile;
+import com.google.android.gnd.model.basemap.tile.Tile.State;
+import com.google.android.gnd.model.feature.Feature;
+import com.google.android.gnd.model.feature.FeatureMutation;
+import com.google.android.gnd.model.feature.Point;
+import com.google.android.gnd.model.form.Element;
+import com.google.android.gnd.model.form.Field;
+import com.google.android.gnd.model.form.Field.Type;
+import com.google.android.gnd.model.form.Form;
+import com.google.android.gnd.model.form.MultipleChoice;
+import com.google.android.gnd.model.form.MultipleChoice.Builder;
+import com.google.android.gnd.model.form.MultipleChoice.Cardinality;
+import com.google.android.gnd.model.form.Option;
+import com.google.android.gnd.model.layer.Layer;
+import com.google.android.gnd.model.layer.Style;
+import com.google.android.gnd.model.observation.Observation;
+import com.google.android.gnd.model.observation.ObservationMutation;
+import com.google.android.gnd.model.observation.ResponseDelta;
+import com.google.android.gnd.model.observation.ResponseMap;
+import com.google.android.gnd.model.observation.TextResponse;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import io.reactivex.subscribers.TestSubscriber;
+import java.util.AbstractCollection;
+import java.util.Date;
+import java8.util.Optional;
+import javax.inject.Inject;
+import org.hamcrest.MatcherAssert;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(application = TestApplication.class)
+public class LocalDataStoreTest {
+
+  // This rule makes sure that Room executes all the database operations instantly.
+  @Rule public InstantTaskExecutorRule instantTaskExecutorRule = new InstantTaskExecutorRule();
+
+  @Inject LocalDataStore localDataStore;
+
+  @Before
+  public void setUp() {
+    DaggerTestComponent.create().inject(this);
+  }
+
+  private User createUser() {
```

Comment:
Please make helpers static, include word "Test" so it's clear these don't just create a new empty instance (e.g. `createTestUser()`).